Common MDI

aciarma opened the pull request:
Added Lumical, quadrupoles and magnetic fields to the common MDI folder for FCCee.

- [OK] Move LumiCal (from CLD_o2_v07) to common folder
- [OK] Add file with dimensions/materials related to MDI elements
- [OK] Add solenoids Fieldmap
- [OK] Add simple quadrupoles (these will change with the lattice, both field and position/dimension)
- [OK] Add a standalone master for MDI elements only (mainly for debug processes and tracking in LumiCal)
- Change detectors master xml to point to the common MDI files, and remove old MDI files from the detector folders
        - CLD (which version?)
        - [OK] IDEA_o1_v03
        - [OK] ALLEGRO_o1_v03

aciarma committed:
Merge pull request #7 from key4hep/main

update from master

aciarma committed:
taken lumical from CLD_o2_v06

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
dimensions compact + standalone compact

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
updated for LumiCal_o3_v05

aciarma committed:
added field maps (to fix) and quadrupoles

aciarma committed:
change name FFQuads

aciarma committed:
fix FFQ file name

aciarma committed:
addded FFQ magnetic field for different optics

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
field map correctly read. fix a variable in FFQ

aciarma committed:
add 2T detector solenoid field

aciarma committed:
fix ideal antisolenoid field

aciarma committed:
apply also to MDI_o1_v01

aciarma committed:
fix identical files

aciarma committed:
fix v01

BrieucF commented on FCCee/MDI/compact/MDI_o1_v00/fields_solenoid.xml:
Hi Andrea, the detector field will be different per detector (it depends e.g. on the coil radius), I think it makes more sense to leave it under the detector specific repositories

aciarma replied:
what if I leave the fields_solenoid.xml in the MDI repo, and put the variables in the DectDimensions ?

BrieucF replied:
I think it will be puzzling for people if the detector field is defined in the MDI folder, even if its parameters are set in the detector specific area. If you are relying on some naming conventions, maybe you could put a comment next to the relevant parameters in the detector xml stating "Do not modify this name, it is needed for MDI" equivalent. If not, what is the reason behind this?

aciarma committed:
update README

aciarma committed:
update README

aciarma committed:
add MDI elements to ALLEGRO_o1_v03

aciarma committed:
update on ALLEGRO_o1_v03

aciarma committed:
add MDI elements to IDEA_o1_v03

aciarma committed:
fixed paths in IDEA test

aciarma committed:
fix IDEA_withDRC test

aciarma commented:
I've changed the master compact file for IDEA_o1_v03 and ALLEGRO_o1_v03 to point to the common MDI folder for all the related elements, removed the files for LumiCal/beam pipe/beam instrumentation from the experiment folders and modified accordingly the DectDimensions.xml files. 

Should I do the same for CLD? If yes, which version?

aciarma committed:
fix IDEA_withDRC test again

aciarma committed:
FFQuads_v01 made not sensitive, FFQuads_v01_sens to be used when needed

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
fix non sensitive quads

aciarma committed:
fix non sensitive quads

danieljeans commented:
are these MDI models expected to be overlap-free yet? @Victor-Schwan and I think we are seeing several overlaps when using the overlap.mac macros in ddsim on both the MDI_standalone_o1_v01 and MDI_standalone_o1_v00 models.

danieljeans commented:
I propose applying g4 step limits to the interior of the beampipe, and the whole MDI region (in my experience this helps achieve accurate simulation of beamstrahlung (ie low pT charged particles) in non-uniform fields). 
Adding something like:
```
<limitset name="mdi_limits">
   <limit name="step_length_max" particles="*" value="5.0" unit="mm" />
</limitset>
```

and adding the limits to the different MDI detectors, e.g. changing

 `<detector name="BeBeampipe" type="DD4hep_Beampipe_o1_v01" insideTrackingVolume="true" nocore="true" vis="BeamPipeVis">`

to

`<detector name="BeBeampipe" type="DD4hep_Beampipe_o1_v01" insideTrackingVolume="true" limits="mdi_limits" nocore="true" vis="BeamPipeVis">`

should work OK, I think.

BrieucF commented on fieldmaps/FCCee_fieldMapXYZ_120218.root:
Are we sure this map contains only the fields from the anti-solenoid and assume a 0 detector field when produced? Otherwise we will "double count" the detector field there. By the way, "fieldMap" is very generic, can you find a more descriptive name for this file.

atolosadelgado commented:
Hi @aciarma ,

Thank you for making this very complete PR!

Here are few comments:
v00:
1. Overlaps are found with the HOM absorber. However, CLD has no overlaps in its MDI, how is it possible?
```
          Overlap is detected for volume AV_4!BeamPipeShield_assembly_3#3!tube_TaShield_0_0#0:4 (G4SubtractionSolid) with HOMAbsorber_envelope_11:11 (G4UnionSolid)
          overlap at local point (38.0487,-33.0895,1241.63) by 4.41048 mm  (max of 124 cases)
--
          Overlap is detected for volume AV_4!BeamPipeShield_assembly_3#3!tube_TaShield_1_1#1:5 (G4SubtractionSolid) with HOMAbsorber_envelope_11:11 (G4UnionSolid)
          overlap at local point (-37.5385,-32.9308,-1209.05) by 4.5692 mm  (max of 94 cases)
--
          Overlap is detected for volume HOMAbsorber_envelope_11:11 (G4UnionSolid) with AV_4!BeamPipeShield_assembly_3#3!tube_TaShield_0_0#0:4 (G4SubtractionSolid)
          overlap at local point (-12.5365,34.1472,-389.019) by 1.48998 cm  (max of 4272 cases)
--
          Overlap is detected for volume AV_8!HOMAbsorbers_PunchThrough_assembly_12#12!tube_HOM1_0_0#0:13 (G4SubtractionSolid) with AV_4!BeamPipeShield_assembly_3#3!tube_TaShield_0_0#0:4 (G4SubtractionSolid)
          overlap at local point (0.553117,-31.1543,-388.835) by 1.96526 cm  (max of 5628 cases)
--
          Overlap is detected for volume AV_8!HOMAbsorbers_PunchThrough_assembly_12#12!tube_HOM1_1_1#1:14 (G4SubtractionSolid) with AV_4!BeamPipeShield_assembly_3#3!tube_TaShield_1_1#1:5 (G4SubtractionSolid)
          overlap at local point (0.785344,-31.6609,-388.835) by 1.96071 cm  (max of 5595 cases)
```
2. Variable QD0 min/max are defined twice, can we centralize it into the Dimensions and material compact file?
3. These variables QD0 have a comment that says potentially outdated, do you know if it means that the value is not valid anymore?
4. Centralize size variables from solenoid and antisolenoid in Dimensions and Materials compact file?
5. Move the definition of magnetic field of FFQ to another file, to be coherent with the solenoid/antisolenoid elements?

v01: 
6. CAD design corresponding to water and AlBeMet volumes seem to have at least 1 facet with the wrong orientation, as reported during initializing the geometry:
```
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomSolids1001
      issued by : G4TessellatedSolid::SetSolidClosed()
Defects in solid: "Componente1"_shape_0xd5bd950 - some facets have wrong orientation!
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------


-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomSolids1001
      issued by : G4TessellatedSolid::SetSolidClosed()
Defects in solid: "Componente1"_shape_0xe064510 - some facets have wrong orientation!
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
```
7. I am not sure if it is related to the previous point, but the Geant4 navigator reports error around point r~25, z=+/-1111 (I think the units are mm). Please see the the attached file named `MDI_v01_error_points.txt`
[MDI_v01_error_points.txt](https://github.com/user-attachments/files/17415295/MDI_v01_error_points.txt)

8. Regarding Daniel comment about region limits, I can take care of it in a future PR.
9. Regarding your question if CLD should be modified, we can leave it as it is for the moment, and let the CLD responsibles change to the central MDI.

Please, let me know if I can help somehow.

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
DD4hep_Mask for central chamber, CAD updates, beam inside pipe

aciarma commented:
New CAD models now have the latest description for elliptoconical chamber cooling manifolds, and the flange at the end of the albemet part
<img width="738" alt="image" src="https://github.com/user-attachments/assets/65d22a7a-7525-4709-ab7a-3d1ff2f4e36e" />

Added preliminar model for Bellows
<img width="282" alt="image" src="https://github.com/user-attachments/assets/f6d42f57-2ba6-45e4-92c8-f15d7dbb04d2" />

Central chamber from ±90mm is now made using DD4hep_Mask instead of CAD to remove issues due to mismatch of facets of the different volumes

Added "beam" volume fitting the CAD beam pipe model to fix air in beam pipe
<img width="632" alt="image" src="https://github.com/user-attachments/assets/61dc64bf-d1e4-40d6-aeb6-5d0ab77201e9" />

Tungsten SR mask reworked, and the notch on the other side of the beam pipe has been removed.


How can I upload the STL files for the new CAD models?

atolosadelgado commented:
> New CAD models now have the latest description for elliptoconical chamber cooling manifolds, and the flange at the end of the albemet part <img alt="image" width="738" src="https://private-user-images.githubusercontent.com/92432372/412082696-65d22a7a-7525-4709-ab7a-3d1ff2f4e36e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyOTY1NjcsIm5iZiI6MTczOTI5NjI2NywicGF0aCI6Ii85MjQzMjM3Mi80MTIwODI2OTYtNjVkMjJhN2EtNzUyNS00NzA5LWFiN2EtM2QxZmYyZjRlMzZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjExVDE3NTEwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3M2M3MzQwYzU0ODgxZTM4MGY1YWIzOWM3MjU5ZTNiMjQ1OGVjNmNjYjlmNGFkMjQ1OTFhYjY2MjA4NDMxOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TFDXp-_OfAFi-z_sfNwnO_UmHTIiXAOEOyW-CkiTN4M">
> 
> Added preliminar model for Bellows <img alt="image" width="282" src="https://private-user-images.githubusercontent.com/92432372/412083108-f6d42f57-2ba6-45e4-92c8-f15d7dbb04d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyOTY1NjcsIm5iZiI6MTczOTI5NjI2NywicGF0aCI6Ii85MjQzMjM3Mi80MTIwODMxMDgtZjZkNDJmNTctMmJhNi00NWU0LTkyYzgtZjE1ZDdkYmIwNGQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjExVDE3NTEwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzMzMyNGM0YjI2OGI5MmRiOTFiMjkwNDEyMDZkN2Q5NTI3OTZjOGY4YTU3ZGVhYmVkYjUzYmJiN2M1ZDAzYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Wx7a-_qRBchnBoekwcfSiEyVOfrHb9RgQe9bWfbSkEE">
> 
> Central chamber from ±90mm is now made using DD4hep_Mask instead of CAD to remove issues due to mismatch of facets of the different volumes
> 
> Added "beam" volume fitting the CAD beam pipe model to fix air in beam pipe <img alt="image" width="632" src="https://private-user-images.githubusercontent.com/92432372/412083844-61dc64bf-d1e4-40d6-aeb6-5d0ab77201e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyOTY1NjcsIm5iZiI6MTczOTI5NjI2NywicGF0aCI6Ii85MjQzMjM3Mi80MTIwODM4NDQtNjFkYzY0YmYtZDFlNC00MGQ2LWFlYjYtNWQwYWI3NzIwMWU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjExVDE3NTEwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhOGRlNmNlMjg3OTE0OTA1NDU3MjY4YWU4NTQ0NmNiYzAwYWQ5NGYzOWM0OGRhMDhlMTYzMzJkOTllMzJkODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6YPpsZFHbgWnpiH450-Cbo6HkEWcxHucPR9X-EcG1cI">
> 
> Tungsten SR mask reworked, and the notch on the other side of the beam pipe has been removed.
> 
> How can I upload the STL files for the new CAD models?

Thank @aciarma for this PR. I am checking your files before moving them to www-eos. I have some questions about the new model:

1. Can you fix this error?
```
Evaluator        WARN  +++ Overwriting variable: name=QD0_min_z value=2000*mm  Evaluator::Object : existing variable [setVariable error]
Detector         WARN  +++ Object 'QD0_min_z' is already defined. New value will be ignored
Evaluator        WARN  +++ Overwriting variable: name=QD0_max_z value=5400*mm  Evaluator::Object : existing variable [setVariable error]
Detector         WARN  +++ Object 'QD0_max_z' is already defined. New value will be ignored
```
2. Can you confirm that the placement of 4 lumicals is expected?
```
dd4hep::Ref_t create_detector(dd4hep::Detector&, xml_h, dd4hep::SensitiveDetector)
Here is my LumiCal
 and this is the sensitive detector: 0x7ffdf517f868
 The crossing angle is: 0.03 radian
Total Length 11.25 cm
Layer Thickness 0.45 cm
Compact          INFO  ++ Converted subdetector:LumiCal of type LumiCal_o1_v01 [calorimeter]
dd4hep::Ref_t create_detector(dd4hep::Detector&, xml_h, dd4hep::SensitiveDetector)
Here is my LumiCal
 and this is the sensitive detector: 0x7ffdf517f868
 The crossing angle is: 0.03 radian
Total Length 11.25 cm
Layer Thickness 0.45 cm
Compact          INFO  ++ Converted subdetector:LumiCalInstrumentation of type LumiCal_o1_v01 [calorimeter]
dd4hep::Ref_t create_detector(dd4hep::Detector&, xml_h, dd4hep::SensitiveDetector)
Here is my LumiCal
 and this is the sensitive detector: 0x7ffdf517f868
 The crossing angle is: 0.03 radian
Total Length 11.25 cm
Layer Thickness 0.45 cm
Compact          INFO  ++ Converted subdetector:LumiCalCooling of type LumiCal_o1_v01 [calorimeter]
dd4hep::Ref_t create_detector(dd4hep::Detector&, xml_h, dd4hep::SensitiveDetector)
Here is my LumiCal
 and this is the sensitive detector: 0x7ffdf517f868
 The crossing angle is: 0.03 radian
Total Length 0.35 cm
Layer Thickness 0.35 cm
Compact          INFO  ++ Converted subdetector:LumiCalBackShield of type LumiCal_o1_v01 [calorimeter]
```

3. I noticed that there are some overlaps, up to 9mm, could you please take a look to it?
```
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_8!Beampipe_STL_0#0!ASCII_0#0:0 (G4TessellatedSolid) with AV_14!Beam_STL_6#6!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (14.0421,-8.71044,659.727) by 3.37392 mm  (max of 1115 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_14!Beam_STL_6#6!ASCII_0#0:0 (G4TessellatedSolid) with AV_8!Beampipe_STL_0#0!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (14.6671,-12.8952,862.223) by 2.16673 mm  (max of 2759 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_15!Paraffin_STL_7#7!ASCII_0#0:0 (G4TessellatedSolid) with AV_8!Beampipe_STL_0#0!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (12.416,-3.81531,117.483) by 506.795 um  (max of 4403 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_16!Water_STL_8#8!ASCII_0#0:0 (G4TessellatedSolid) with AV_8!Beampipe_STL_0#0!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (-18.312,11.0271,681.066) by 36.7902 um  (max of 2395 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_17!Bellows_STL_9#9!ASCII_0#0:0 (G4TessellatedSolid) with AV_8!Beampipe_STL_0#0!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (8.8925,-47.5236,1225.23) by 8.56756 mm  (max of 52 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_18!Beampipe_crotch_STL_10#10!ASCII_0#0:0 (G4TessellatedSolid) with AV_14!Beam_STL_6#6!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (-14.8662,-12.1879,-1323.04) by 944.42 um  (max of 3559 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
--
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomVol1002
      issued by : G4PVPlacement::CheckOverlaps()
Overlap with volume already placed !
          Overlap is detected for volume AV_19!Tungsten_STL_11#11!ASCII_0#0:0 (G4TessellatedSolid) with AV_14!Beam_STL_6#6!ASCII_0#0:0 (G4TessellatedSolid)
          overlap at local point (-44.6705,1.14337,2069.84) by 3.08147 um  (max of 1403 cases)
NOTE: Reached maximum fixed number -1- of overlaps reports for this volume !
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
```

4. I think the complaint about the facets with wrong orientation persists, could you confirm if the shapes giving this problem before are the same?
```
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomSolids1001
      issued by : G4TessellatedSolid::SetSolidClosed()
Defects in solid: ASCII_shape_0xdde5530 - some facets have wrong orientation!
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------


-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomSolids1001
      issued by : G4TessellatedSolid::SetSolidClosed()
Defects in solid: ASCII_shape_0xf00fe90 - some facets have wrong orientation!
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
```

aciarma committed:
removed double definition of QD0_min_z and QD0_max_z

aciarma committed:
removed double definition of QD0_min_z and QD0_max_z

aciarma commented:
hello @atolosadelgado, on the above points:

1. I fixed this
2. It should be correct because they are 4 different parts of LumiCal (main, Instrumentation, Cooling and BackShield), but I did not do anything on that, I just moved the latest version in the MDI folder
3. on my side if I do `checkOverlap -c MDI_standalone_o1_v01.xml` I get no overlaps, how did you get those warnings?
4. on the facets orientation, I did not understood why this comes up... from CAD it does not seems that there are such problems, and I also tried to use tools I found online (trimesh python package) but it had no effect apparently.

atolosadelgado commented:
> hello @atolosadelgado, on the above points:
> 
>     1. I fixed this
> 
>     2. It should be correct because they are 4 different parts of LumiCal (main, Instrumentation, Cooling and BackShield), but I did not do anything on that, I just moved the latest version in the MDI folder
> 
>     3. on my side if I do `checkOverlap -c MDI_standalone_o1_v01.xml` I get no overlaps, how did you get those warnings?
> 
>     4. on the facets orientation, I did not understood why this comes up... from CAD it does not seems that there are such problems, and I also tried to use tools I found online (trimesh python package) but it had no effect apparently.

Hi @aciarma 

thanks a lot for your quick response! The wrong facet orientation, if I remember correctly, happened at z=+/-1111mm. I will check again running with geantinos. 

Regarding the overlaps, I run first few events with geantinos and Geant4 tracking already complained about it. Then, I check the overlaps using Geant4 with the following command:
```
ddsim --compactFile FCCee/MDI/compact/MDI_o1_v01/MDI_standalone_o1_v01.xml --runType run --macroFile utils/overlap.mac > overlapdump.txt
```

Please let me know if I can help further. Thanks for your quick reply!

aciarma commented:
Thank you, I see the same overlaps now. We are investigating a way to avoid this but it seems not trivial. Geometries are obtained from CAD where there are no overlaps, but once it is translated to STL they appear due to the facets not being aligned, and boolean operations on meshes are not working nicely. I will follow up on this.

atolosadelgado commented:
> Thank you, I see the same overlaps now. We are investigating a way to avoid this but it seems not trivial. Geometries are obtained from CAD where there are no overlaps, but once it is translated to STL they appear due to the facets not being aligned, and boolean operations on meshes are not working nicely. I will follow up on this.
> 
> (this is also why I decided to use DD4hep shapes for the central chamber)

Thank you very much for crosschecking! Please let me know if I can help further :)

aciarma commented:
We are trying different export strategies using STL, still we are observing small overlaps, now ~20um. 
We are thinking of tapering a little bit the light materials to leave a small tolerance between touching volumes (would be filled by world volume air). Do you think it would be an acceptable solution?

AutoCAD Inventor can export also in OBJ of the ddsim supported formats, but when I try to run the overlap macro on the test file [](https://github.com/AIDASoft/DD4hep/blob/master/examples/DDCAD/compact/Check_Shape_OBJ_spider.xml) I get a crash. Is OBJ still supported? Or are there good formats other than those in the DDCAD/examples list?

aciarma commented:
ok I realized that the spider.obj file called by the OBJ example in DDCAD does not actually exist. Anyway, following that example I tried to read a simple OBJ file we generated with AutoCAD, 

```
<detector id="1" name="Shape_OBJ" type="DD4hep_TestShape_Creator">
      <check vis="AlBeMet_vis">
        <shape type="CAD_MultiVolume" ref="${K4GEO}/FCCee/MDI/compact/MDI_o1_v01/stl_files/PROVE/prova1_stlOBJ.obj" unit="mm">
          <volume id="0"  name="v1" material="Gold" vis="AlBeMet_vis"/>
         </shape>
      </check>
    </detector>
    ```

but got the following error

```
Compact          ERROR ++ FAILED    to convert subdetector: Shape_OBJ: Cannot find a material referenced by name:191,191,191
```

where 191,191,191 I highly suspect it is the RGB color code of the volume (a light grey).

Is there a way to overwrite such fields? I defined Gold as material but it clearly did not work.
Also, why the [Check_Shapes_OBJ_spider.xml](https://github.com/AIDASoft/DD4hep/blob/master/examples/DDCAD/compact/Check_Shape_OBJ_spider.xml) exists but the .obj file does not? There was some problem with that format?

atolosadelgado commented:
> We are trying different export strategies using STL, still we are observing small overlaps, now ~20um. We are thinking of tapering a little bit the light materials to leave a small tolerance between touching volumes (would be filled by world volume air). Do you think it would be an acceptable solution?
> 
> AutoCAD Inventor can export also in OBJ of the ddsim supported formats, but when I try to run the overlap macro on the test file I get a crash. Is OBJ still supported? Or are there good formats other than those in the DDCAD/examples list?

Hi @aciarma 

Sorry for the late reply! I believe DDCAD is built on top of ASSIMP, so it should support many formats [1].

If you could share the OBJ file privately, I'd be happy to take a look at it. 

[1] https://github.com/assimp/assimp/blob/master/doc/Fileformats.md

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
Merge branch 'key4hep:main' into master

aciarma committed:
removed common lumical for now

aciarma committed:
removed common lumical for now

aciarma committed:
Revert "removed common lumical for now"

This reverts commit 44f91d7e66096dab24d3c38af9928c144bcfa8f3.

aciarma committed:
Revert "removed common lumical for now"

This reverts commit c6313207ac3c8da0410ca59c8e5cda64ec04a30d.

aciarma committed:
fixed common lumical inconsistencies

aciarma committed:
fixed common lumical inconsistencies

BrieucF commented:
Hi Andrea, thanks for reviving this. I thought it would have been easier to close this one and tackle everything it does in smaller PR's but ok, let's proceed. Some questions/comments:
 - Are the new STL files also downloaded with the flag `INSTALL_BEAMPIPE_STL_FILES=ON` ?
 - Can we avoid having multiple version of the MDI in a single MDI_o*_v* folder? It is probably enough to just create a new MDI folder to host the new CAD beampipe.
 - can you answer [this](https://github.com/key4hep/k4geo/pull/381#discussion_r1792924067) question please?
 - IIUC, this PR won't change the physics (we still load the shape based beampipe by default, which is ok, and it has not changed w.r.t. previous version, right?) except for the change in the magnetic field from the MDI components. In order to avoid changing silently the physics, I'd propose to keep the analytical MDI fields by default, but to leave the loading of the field map commented out (so that it is easy for the user to enable it).  
 - can you apply the changes you made to ALLEGRO_o1_v03 and IDEA_o1_v03 also to IDEA_o2_v01 and ALLEGRO_o2_v01?

aciarma committed:
fix lumical variables definitions

aciarma committed:
Merge branch 'key4hep:main' into commonMDI

aciarma committed:
work in progress

aciarma committed:
common MDI for IDEA_o1_v03

aciarma committed:
common MDI for IDEA_o2_v01

BrieucF commented:
I guess that could work as well, but then you would keep e.g. the lumiCal in the folder?
